databricks/koalas/indexes.py

```diff
@@ -1546,6 +1554,97 @@ def argmin(self):
 
         return sdf.orderBy(self.spark.column.asc(), F.col(sequence_col).asc()).first()[0]
 
+    def putmask(self, mask, value):
+        """
+        Return a new Index of the values set with the mask.
+
+        .. note:: this API can be pretty expensive since it is based on
+             a global sequence internally.
+
+        Returns
+        -------
+        Index
+
+        Example
+        -------
+        >>> kidx = ks.Index([1, 2, 3, 4, 5])
+        >>> kidx
+        Int64Index([1, 2, 3, 4, 5], dtype='int64')
+
+        >>> kidx.putmask(kidx > 3, 100).sort_values()
+        Int64Index([1, 2, 3, 100, 100], dtype='int64')
+
+        >>> kidx.putmask(kidx > 3, ks.Index([100, 200, 300, 400, 500])).sort_values()
+        Int64Index([1, 2, 3, 400, 500], dtype='int64')
+        """
+        scol_name = self._internal.index_spark_column_names[0]
+        sdf = self._internal.spark_frame.select(self.spark.column)
+
+        dist_sequence_col_name = verify_temp_column_name(sdf, "__distributed_sequence_column__")
+        sdf = InternalFrame.attach_distributed_sequence_column(
+            sdf, column_name=dist_sequence_col_name
+        )
+
+        replace_col = verify_temp_column_name(sdf, "__replace_column__")
+        masking_col = verify_temp_column_name(sdf, "__masking_column__")
+
+        if isinstance(value, (list, tuple, Index, Series)):
+            if isinstance(value, (list, tuple)):
+                pandas_value = pd.Series(value)
+            elif isinstance(value, (Index, Series)):
+                pandas_value = value.to_pandas()
+
+            if self.size != pandas_value.size:
+                # TODO: We can't support different size of value for now.
+                raise ValueError("value and data must be the same size")
+
+            replace_return_type = as_spark_type(pandas_value.dtype.type)
```

Comment:
Let's try to use `infer_pd_series_spark_type` rather than `as_spark_type`.

You can like as below.

```python
replace_return_type = infer_pd_series_spark_type(pandas_value)
```

Reply:
thank you! :) I fixed it by using `infer_pd_series_spark_type `